sql-plugin/src/main/scala/org/apache/spark/sql/rapids/HashFunctions.scala

```diff
@@ -74,6 +74,101 @@ case class GpuSha1(child: Expression)
   }
 }
 
+/**
+ * GPU implementation of the SHA2 hash function set.
+ * Note:
+ *   - Only bit lengths of 0, 224, 256, 384, and 512 are supported.
+ *   - A bit length of 0 is treated as 256, following Spark behaviour.
+ *   - For the GPU implementation, the bit length must be provided as a literal.
+ *     This differs from the CPU implementation, where the bit length can differ
+ *     per row.
+ * @see https://spark.apache.org/docs/latest/api/sql/index.html#sha2
+ * @param left The BINARY input column to be hashed.
+ * @param right The INTEGER literal bitlength for hashing.
+ */
+case class GpuSha2(left: Expression, right: Expression)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerantShim {
+  private val childExpr: Expression = left
+  private val bitLengthExpr: Expression = right
+  override def toString: String = s"sha2($childExpr, $bitLength)"
+  override def inputTypes: Seq[AbstractDataType] = Seq(BinaryType, IntegerType)
+  override def dataType: DataType = StringType
+
+  private val bitLength: Option[Int] = bitLengthExpr match {
+    case lit: GpuLiteral if lit.value.isInstanceOf[Int] =>
+      lit.value.asInstanceOf[Int] match {
+        case 224 | 256 | 384 | 512 => Some(lit.value.asInstanceOf[Int])
+        case 0 => Some(256) // Spark default
+        case _ => None // SHA2 returns null for other bit lengths.
+      }
+    case _ => None
+  }
+
+  override def nullable: Boolean = childExpr.nullable || bitLength.isEmpty
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    if (bitLength.isEmpty) {
+      // SHA2 returns null for "unsupported" bit lengths.
+      withResource(Scalar.fromNull(DType.STRING)) { nullStringExemplar =>
+        ColumnVector.fromScalar(nullStringExemplar, lhs.getRowCount.toInt)
+      }
+    } else {
+        bitLength match {
```

Comment:
NIT: The indent is expected as only two spaces, but it is 4 here.

Reply:
This was the result of removing an enclosing scope.  Fixed.